fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/impl/object/builder/TradeOfferInternals.java

```diff
@@ -0,0 +1,121 @@
+	private static final Map<VillagerProfession, Int2ObjectMap<TradeOffers.Factory[]>> VILLAGER_TRADE_FACTORIES = new HashMap<>();
+	private static final Int2ObjectMap<TradeOffers.Factory[]> WANDERING_TRADER_FACTORIES = new Int2ObjectOpenHashMap<>();
+	private TradeOfferInternals() {
+	}
+
+	public static void registerVillagerOffers(VillagerProfession profession, int level, Consumer<List<TradeOffers.Factory>> factory) {
+		final List<TradeOffers.Factory> list = new ArrayList<>();
+		factory.accept(list);
+
+		final TradeOffers.Factory[] additionalEntries = list.toArray(new TradeOffers.Factory[0]);
+		final Int2ObjectMap<TradeOffers.Factory[]> professionEntry = VILLAGER_TRADE_FACTORIES.computeIfAbsent(profession, p -> new Int2ObjectOpenHashMap<>());
+
+		final TradeOffers.Factory[] currentEntries = professionEntry.computeIfAbsent(level, l -> new TradeOffers.Factory[0]);
+		final TradeOffers.Factory[] newEntries = ArrayUtils.addAll(additionalEntries, currentEntries);
+		professionEntry.put(level, newEntries);
+
+		// Refresh the trades map
+		TradeOfferInternals.refreshOffers();
+	}
+
+	public static void registerWanderingTraderOffers(int level, Consumer<List<TradeOffers.Factory>> factory) {
+		final List<TradeOffers.Factory> list = new ArrayList<>();
+		factory.accept(list);
+
+		final TradeOffers.Factory[] additionalEntries = list.toArray(new TradeOffers.Factory[0]);
+		final TradeOffers.Factory[] currentEntries = TradeOfferInternals.DEFAULT_WANDERING_TRADER_OFFERS.computeIfAbsent(level, key -> new TradeOffers.Factory[0]);
+
+		// Merge current and new entries
+		final TradeOffers.Factory[] newEntries = ArrayUtils.addAll(additionalEntries, currentEntries);
+		TradeOfferInternals.DEFAULT_WANDERING_TRADER_OFFERS.put(level, newEntries);
+
+		// Refresh the trades map
+		TradeOfferInternals.refreshOffers();
+	}
+
+	public static void refreshOffers() {
+		TradeOfferInternals.refreshVillagerOffers();
+		TradeOfferInternals.refreshWanderingTraderOffers();
+	}
+
+	private static void refreshVillagerOffers() {
+		final HashMap<VillagerProfession, Int2ObjectMap<TradeOffers.Factory[]>> trades = new HashMap<>(TradeOfferInternals.DEFAULT_VILLAGER_OFFERS);
+
+		for (Map.Entry<VillagerProfession, Int2ObjectMap<TradeOffers.Factory[]>> tradeFactoryEntry : TradeOfferInternals.VILLAGER_TRADE_FACTORIES.entrySet()) {
+			// Create an empty map or get all existing profession entries.
+			final Int2ObjectMap<TradeOffers.Factory[]> leveledFactoryMap = trades.computeIfAbsent(tradeFactoryEntry.getKey(), k -> new Int2ObjectOpenHashMap<>());
+			// Get the existing entries
+			final Int2ObjectMap<TradeOffers.Factory[]> value = tradeFactoryEntry.getValue();
+
+			// Iterate through the existing level entries
+			for (int level : value.keySet()) {
+				final TradeOffers.Factory[] factories = value.get(level);
+
+				if (factories != null) {
+					ArrayUtils.addAll(leveledFactoryMap.computeIfAbsent(level, key -> new TradeOffers.Factory[0]), factories);
```

Comment:
What, you should store the result of this `addAll` call

Reply:
I think I got it to store now